ci/telemetry_autoscaling.yml

```diff
@@ -0,0 +1,95 @@
+---
+- name: Telemetry autoscaling tests
+  hosts:  "{{ cifmw_target_hook_host | default('localhost')  }}"
+  gather_facts: true
+  vars:
+    fvt_dir: "{{ ansible_env.HOME }}/{{ zuul.projects['github.com/infrawatch/feature-verification-tests'].src_dir }}"
+    ci_framework_logs: "{{ ansible_env.HOME }}/ci-framework-data/logs"
+    openstack_cmd: "oc rsh openstackclient openstack"
+    metrics_backend: "prometheus"
+    stack_image: "cirros"
+    stack_flavor: "m1.small"
+    stack_network: "private"
+    stack_external_network: "public"
+    stack_name: "vnf"
+  environment:
+    KUBECONFIG: "{{ cifmw_openshift_kubeconfig }}"
+    PATH: "{{ cifmw_path }}"
+  tasks:
+    - name: Run the autoscaling test
+      block:
+        - name: Create log dir for autoscaling test
+          ansible.builtin.file:
+            path: "{{ ci_framework_logs }}/functional-tests/autoscaling"
+            state: directory
+            mode: "0755"
+
+         # NOTE: the tags are for testing/development, eventually, the role will just be imported, and main.yml will run through the tasks
+        - import_role:
+            name: telemetry_autoscaling
+            tasks_from: verify_autoscaling
+          tags:
+            - precheck
+
+        - import_role:
+            name: telemetry_autoscaling
+            tasks_from: configure_heat
+          tags:
+            - setup
+
+        - import_role:
+            name: telemetry_autoscaling
+            tasks_from: creating_stack
+          tags:
+            - create
+
+        - import_role:
+            name: telemetry_autoscaling
+            tasks_from: test_autoscaling
+          tags:
+            - test
```

Comment:
This should be replaced with just the one import_role task.

When we add the logging tests, we can have a list of functional_tests (defaults to a list for all roles to run for OSP18).
Each of the test roles we want to run will be included here, and each will be guarded with `when: "test_autoscaling" in functional_tests`

```
- import_role:
     name: telemetry_autoscaling
   when: '"telemetry_autoscaling" in functional_tests'

- import_role:
      name: test_logging
   when: '"test_logging" in functional_tests' 
 ```

Or we can use a with_items: <functional_tests>, and import_role: item
e.g. 
```
- import_role:
    name: "{{ item }}"
  with_items: "{{ functional_tests }}"
```

Reply:
I was thinking to have a separate playbook for each test , and just call it from the run_playbooks but the idea to have them as a list is much better in scaling and we can rename the telemetry_autoscaling.yml to something more generic to include all the telemetry_tests with osp18